name/registry.go

```diff
@@ -23,6 +23,7 @@ const (
 
 // Registry stores a docker registry name in a structured form.
 type Registry struct {
+	Scheme   string
```

Comment:
I don't really like that this is public / mutable.

I wonder if we should simply move `transport.Scheme` to be a property of Registry, and add a `NewInsecureRegistry` constructor.

Reply:
Looks good! I will try this.